### ISSUE TITLE
Potential preprocessing inconsistency when adding global evaluation

### ISSUE THREAD

**Issue by dafnePap** [KarhouTam/SCAFFOLD-PyTorch#6]
This is not an issue for the current code as-is, since client-side training, validation, and testing are all internally consistent.

However, the behavior becomes problematic when introducing a global test set on the server side (like using torchvision MNIST for centralized evaluation).

In data/utils/run.py, transforms.Normalize(mean, std) is applied during dataset partitioning without explicitly ensuring that the input has first been scaled to [0,1] (e.g. via ToTensor or division by 255). When applied directly on raw MNIST pixel values (0–255), this results in extremely large input magnitudes.

Example debug output from a client batch:
[DEBUG CLIENT 8] min -0.43349921703338623 max 845.337646484375 mean 104.27739715576172 std 253.90936279296875

Because client train/val/test splits are all generated from the same pipeline, client-side accuracies appear to be okay.
However, if a global server-side test set is added using standard torchvision preprocessing (ToTensor + Normalize), a distribution mismatch occurs,
leading to misleading global evaluation results.

This could likely be avoided by ensuring inputs are scaled to [0,1] before Normalize, or by handling scaling and normalization directly in the dataset class instead of relying on stored transforms.

**Comment by KarhouTam** [comment]
Sorry for the late reply. Thanks for your attention to this repo first. However, I almost forgot the details in this codebase, sorry about that 😂 

If available, welcome to transfer to [FL-bench](https://github.com/KarhouTam/FL-bench/tree/master), which also contains SCAFFOLD method and has stronger performance and features (like parallelism), also, still maintained.

I am gonna archive this repo soon.
